src/contexts/search-context.ts

```diff
@@ -26,14 +26,15 @@ export function createSearchContext(
     hitsPerPage: instantSearchParams?.hitsPerPage,
     page: instantSearchParams?.page,
   })
+
   const searchContext: SearchContext = {
     ...options,
     ...instantSearchParams,
     sort: sortByArray.join(':') || '',
     indexUid,
     pagination,
     defaultFacetDistribution,
-    placeholderSearch: !options.placeholderSearch, // true by default
+    placeholderSearch: options.placeholderSearch !== false, // true by default
```

Comment:
This option is protected by type? Because if the user add `123`, `0` in this option, the option will be `true` :)